src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java

```diff
@@ -60,28 +60,27 @@ void configureMavenExecution(MavenExecutionRequest request, List<RepoFromEnv> re
     
             for (RepoFromEnv repoFromEnv : reposFromEnv) {
     
-                Repository repository = getRepository(repoFromEnv);
+                repositoriesFromEnv.addRepository(getRepository(repoFromEnv));
+                
+                repositoriesFromEnv.addPluginRepository(getRepository(repoFromEnv));
     
-                repositoriesFromEnv.addRepository(repository);
-    
-                if (repoFromEnv.getUsername() != null && repoFromEnv.getPassword() != null) {
-                    Server server = new Server();
-                    server.setId(repoFromEnv.getId());
-                    server.setUsername(repoFromEnv.getUsername());
-                    server.setPassword(repoFromEnv.getPassword());
-                    request.addServer(server);
+                if (repoFromEnv.getUsername() != null) {
+                    request.addServer(getServer(repoFromEnv));
                 }
-                logger.debug("Repository from system env: " + repoFromEnv.getUrl() + " (id: "+repoFromEnv.getId() 
+                
+                logger.info("Repository from system env: " + repoFromEnv.getUrl() + " (id: "+repoFromEnv.getId() 
```

Comment:
"Additional repository added from system variables ...."

IMHO either "system variables" or "environment variables" (both mean the same here). "System env" feel unclear to me.

Reply:
Let's do this in the logging PR